src/lv_misc/lv_utils.c

```diff
@@ -72,6 +72,42 @@ char * lv_utils_num_to_str(int32_t num, char * buf)
     return buf;
 }
 
+/**
+ * Convert a fixed point number to string
+ * @param num a number
+ * @param number of digits after decimal point
+ * @param buf pointer to a `char` buffer. Buffer must be big enough for number.
+ * @return same as `buf` (just for convenience)
+ */
+char * lv_utils_num_to_str_fixed(int32_t num, int decimals, char * buf)
+{
+    uint16_t dec_power = 1;
+    for(int i=0; i<decimals; i++)
+        dec_power *= 10;
+
+    lv_utils_num_to_str(num / dec_power, buf);
```

Comment:
lv_utils_num_to_str_fixed needs the buffer size
`char * lv_utils_num_to_str_fixed(int32_t num, int32_t decimals, char * buf, size_t bufsize)`

This works until you get trailing zeros.  I'll keep looking at it.  It would need range checking too.
```
lv_snprintf(buf, bufsize, "%d", num);
lv_txt_ins(buf, strlen(buf) - decimals, ".");
```

Reply:
It should be `lv_txt_ins(buf, buf_size - decimals, ".");`, isn't it?

Reply:
The has to be some zero padding for values less than the base 10 exponent.  Also special case for no fixed point.
```
lv_snprintf(buf, bufsize, "%0*d", decimals+1, num);
if(decimals > 0)
    lv_txt_ins(buf, strlen(buf) - decimals, ".");
```
Unfortunately I do not think there is a way to limit the result length for a decimal (for a string there is).

Reply:
lv_snprintf takes care of the maximum buffer size